### PR TITLE
introduce a new set APIs to create/update/delete page sections

### PR DESCRIPTION
**Pull request by sangle7**
introduce a new set APIs to create/update/delete page sections 
Roadmap # 15124

### CODE REVIEW

File: api-reference/beta/api/horizontalsection-delete.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+## Permissions
+
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+>**Note:** To delete an item, the user must have granted the application write access to the item to be deleted.
+
+|Permission type      | Permissions (from least to most privileged)              |
+|:--------------------|:---------------------------------------------------------|
+|Delegated (work or school account) | Sites.ReadWrite.All    |
```
Comment (liun27) [inline api-reference/beta/api/horizontalsection-delete.md:29]: align the "|"?

File: api-reference/beta/api/horizontalsection-create.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+```http
+POST /sites/{site-id}/pages/{page-id}/canvasLayout/horizontalSections
+```
+
+## Request headers
+
+| Name          | Description                 |
+| :------------ | :-------------------------- |
+| Authorization | Bearer {token}. Required.   |
+| Content-Type  | application/json. Required. |
```
Comment (liun27) [inline api-reference/beta/api/horizontalsection-create.md:43]: I am not sure whether we should use `pplication/json;odata.metadata=none` here?

File: api-reference/beta/api/verticalsection-get.md [hunk 3]
```diff
@@ -175,14 +173,11 @@ HTTP/1.1 200 OK
...
-    "emphasis": "neutral",
-    "webparts":[{
-      "@odata.type": "#microsoft.graph.textWebPart",
-      "id": "1212fc8d-dd6b-408a-8d5d-9f1cc787efbb",
-      "innerHtml": "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque vel.</p>"
-    }]
-  }
+  "emphasis": "neutral",
+  "webparts":[{
+    "@odata.type": "#microsoft.graph.textWebPart",
```
Comment (liun27) [inline api-reference/beta/api/verticalsection-get.md:178]: do we still want to document odata here?

File: api-reference/beta/api/verticalsection-update.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+}
+-->
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "emphasis": "strong",
+  "webparts":[
```
Comment (liun27) [inline api-reference/beta/api/verticalsection-update.md:97]: will response carry the body if no expand option provided?

File: api-reference/beta/api/horizontalsection-create.md (hunk 2 above)
Comment (sangle7) [reply api-reference/beta/api/horizontalsection-create.md:43]: No we don't need the header here

**Comment by Danielabom** [comment]
Removed the "content review in progress" label and reached out to the author to get info.

File: api-reference/beta/api/verticalsection-create.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
+---
```
Comment (Danielabom) [inline api-reference/beta/api/verticalsection-create.md:1]: @sangle7, please update the file name accordingly. Here's a link to the guidelines: [https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211751/File-naming]

File: api-reference/beta/api/verticalsection-create.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+title: 'Create verticalSection'
+description: "Create a verticalSection in page"
```
Comment (Danielabom) [inline api-reference/beta/api/verticalsection-create.md:3]:
```suggestion
description: "Create a verticalSection in a page"
```

File: api-reference/beta/api/verticalsection-delete.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
+---
+title: 'Delete a verticalSection'
+description: 'Delete a vertical section from a page.'
+author: sangle7
+ms.localizationpriority: medium
+ms.prod: sharepoint
+doc_type: apiPageType
+---
+
+# Delete a vertical section from a page
```
Comment (Danielabom) [inline api-reference/beta/api/verticalsection-delete.md:10]: Can we leave this as "Delete verticalSection"? Or is there a reason we need to specify that we are deleting a vertical from a page?

File: api-reference/beta/api/verticalsection-delete.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Removes a [verticalSection][] from the [sitePage][]
```
Comment (Danielabom) [inline api-reference/beta/api/verticalsection-delete.md:16]: Please add the links of the resources.

File: api-reference/beta/api/verticalsection-delete.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+[verticalSection]: ../resources/verticalSection.md
```
Comment (Danielabom) [inline api-reference/beta/api/verticalsection-delete.md:18]: Please insert the links in line 16 accordingly.

File: api-reference/beta/api/verticalsection-delete.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+[sitePage]: ../resources/sitepage.md
```
Comment (Danielabom) [inline api-reference/beta/api/verticalsection-delete.md:19]: Please insert the link in line 16 accordingly.

File: api-reference/beta/api/verticalsection-delete.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+|Application | Sites.ReadWrite.All |
+
+## HTTP request
+
+<!-- { "blockType": "ignored" } -->
+
+```http
+DELETE /sites/{site-id}/pages/{page-id}/canvasLayout/verticalSection
+```
+
```
Comment (Danielabom) [inline api-reference/beta/api/verticalsection-delete.md:40]: Please confirm there isn't any info to add in the ## Request headers section.

File: api-reference/beta/api/verticalsection-delete.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+## Request body
+
+Do not supply a request body with this method.
+
+## Response
+
+If successful, this call returns a `204 No Content` response to indicate that the resource was deleted and there was nothing to return.
```
Comment (Danielabom) [inline api-reference/beta/api/verticalsection-delete.md:47]:
```suggestion
If successful, this call returns a `204 No Content`. It does not return anything in the response body.
```

File: api-reference/beta/api/horizontalsection-create.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
+---
```
Comment (Danielabom) [inline api-reference/beta/api/horizontalsection-create.md:1]: Please update the file name accordingly. Here's a link to the guidelines: [https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211751/File-naming]

File: api-reference/beta/api/horizontalsection-create.md [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+title: 'Create horizontalSection'
+description: "Create a horizontalSection in page"
```
Comment (Danielabom) [inline api-reference/beta/api/horizontalsection-create.md:3]:
```suggestion
description: "Create a horizontalSection in a page"
```

File: api-reference/beta/api/horizontalsection-delete.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
+---
+title: 'Delete a horizontalSection'
+description: 'Delete a horizontal section from a page.'
+author: sangle7
+ms.localizationpriority: medium
+ms.prod: sharepoint
+doc_type: apiPageType
+---
+
+# Delete a horizontal section from a page
```
Comment (Danielabom) [inline api-reference/beta/api/horizontalsection-delete.md:10]: Can we leave this as "Delete horizontalSection"? Or is there a reason we need to specify that we are deleting a horizontal section from a page?

File: api-reference/beta/api/horizontalsection-delete.md [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Removes a [horizontalSection][] from the [sitePage][]
```
Comment (Danielabom) [inline api-reference/beta/api/horizontalsection-delete.md:16]: Please insert the links to the resources.  I believe the links are below.

File: api-reference/beta/api/horizontalsection-delete.md [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+| Application                            | Sites.ReadWrite.All                         |
+
+## HTTP request
+
+<!-- { "blockType": "ignored" } -->
+
+```http
+DELETE /sites/{site-id}/pages/{page-id}/canvasLayout/horizontalSections/{horizontalSectionId}
+```
+
```
Comment (Danielabom) [inline api-reference/beta/api/horizontalsection-delete.md:40]: I just want to make sure that we don't need the ## Request headers section. Do we?

**Review by Danielabom — CHANGES_REQUESTED** [review]
Thank you for working on this PR @sangle7. Please review my comments.

File: api-reference/beta/api/horizontalsection-delete.md (hunk 17 above)
Comment (sangle7) [reply api-reference/beta/api/horizontalsection-delete.md:40]: Yes, we won't need specfic headers when calling `delete`

**Comment by Danielabom** [comment]
Hi @sangle7, I wanted to follow up on this PR. Please review my comments, thanks!

**Comment by sangle7** [comment]
> Hi @sangle7, I wanted to follow up on this PR. Please review my comments, thanks!

Thanks @Danielabom. Will take a look soon

File: api-reference/beta/api/horizontalsection-delete.md (hunk 17 above)
Comment (sangle7) [reply api-reference/beta/api/horizontalsection-delete.md:40]: But it looks good to me to have a ## Request headers section mentioned that we only need the 'Authorization' header

File: api-reference/beta/api/verticalsection-delete.md (hunk 11 above)
Comment (sangle7) [reply api-reference/beta/api/verticalsection-delete.md:40]: Updated

File: api-reference/beta/api/horizontalsection-create.md (hunk 13 above)
Comment (sangle7) [reply api-reference/beta/api/horizontalsection-create.md:1]: Thanks, renamed to 'sitepage-post-horizontalsection'

File: api-reference/beta/api/verticalsection-create.md (hunk 5 above)
Comment (sangle7) [reply api-reference/beta/api/verticalsection-create.md:1]: Thanks, renamed to 'sitepage-post-verticalsection'

**Comment by sangle7** [comment]
Hi @Danielabom, I addressed your comments. Could you help to take a look again? Thanks!

**Comment by Danielabom** [comment]
@sangle7, I noticed you've been making changes. Please change the label to "review feedback addressed" as soon as you feel ready. Thanks :)

**Review by Danielabom — APPROVED** [review]
Thank you for working on this PR @sangle7. PR approved.

**Comment by Danielabom** [comment]
@sangle7, just one more thing. There are some build conflicts. Please review.

**Comment by Danielabom** [comment]
@sangle7, friendly reminder to add the "ready to merge" label to this PR as soon as you are ready to publish. Thanks!

**Comment by sangle7** [comment]
Hey folks, We're waiting for code changes reach prod as well as the schema change. It might take longer since we want several features to go alive as the same time.

File: api-reference/beta/api/horizontalsection-delete.md [hunk 18]
```diff
@@ -0,0 +1,70 @@
+---
+title: 'Delete a horizontalSection'
+description: 'Delete a horizontal section from a page.'
+author: sangle7
+ms.localizationpriority: medium
+ms.prod: sharepoint
+doc_type: apiPageType
+---
+
+# Delete horizontal section from a page
```
Comment (Lauragra) [inline api-reference/beta/api/horizontalsection-delete.md:10]: Please change the title and H1 to "Delete horizonalSection".

File: api-reference/beta/api/horizontalsection-delete.md [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Removes a [horizontalSection]( ../resources/horizontalSection.md) from the [sitePage](../resources/sitepage.md).
```
Comment (Lauragra) [inline api-reference/beta/api/horizontalsection-delete.md:16]: Please revise as follows:
Remove a [horizontalSection](../resources/horizontalsection.md) from a [sitePage](../resources/sitepage.md).

File: api-reference/beta/api/horizontalsection-delete.md [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+
+If successful, this call returns a `204 No Content` response to indicate that the resource was deleted and there was nothing to return.
+
+If the target horizontal section is not found, this method returns `404 Not Found`.
+
+## Example
+
+<!-- { "blockType": "request", "name": "delete-horizontalSection", "scopes": "files.readwrite sites.readwrite.all" } -->
+
+##### Request
```
Comment (Lauragra) [inline api-reference/beta/api/horizontalsection-delete.md:58]: Please change the request and response sections to H3s - ###.

File: api-reference/beta/api/verticalsection-delete.md [hunk 21]
```diff
@@ -0,0 +1,69 @@
+---
+title: 'Delete a verticalSection'
+description: 'Delete a vertical section from a page.'
+author: sangle7
+ms.localizationpriority: medium
+ms.prod: sharepoint
+doc_type: apiPageType
+---
+
+# Delete vertical section from a page
```
Comment (Lauragra) [inline api-reference/beta/api/verticalsection-delete.md:10]: Please update the title and H1 to: "Delete verticalSection".

File: api-reference/beta/api/verticalsection-delete.md [hunk 22]
```diff
@@ -0,0 +1,69 @@
...
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Removes a [verticalSection](../resources/verticalSection.md) from the [sitePage](../resources/sitepage.md).
```
Comment (Lauragra) [inline api-reference/beta/api/verticalsection-delete.md:16]: Revise as:
Remove a [verticalSection](../resources/verticalsection.md) from a [sitePage](../resources/sitepage.md).

File: api-reference/beta/api/verticalsection-delete.md [hunk 23]
```diff
@@ -0,0 +1,69 @@
...
+
+If successful, this call returns a `204 No Content`. It does not return anything in the response body.
+
+If the vertical section is not found, this method returns `404 Not Found`.
+
+## Example
+
+<!-- { "blockType": "request", "name": "delete-verticalSection", "scopes": "sites.readwrite.all" } -->
+
+##### Request
```
Comment (Lauragra) [inline api-reference/beta/api/verticalsection-delete.md:58]: Changes these to H3s.

File: api-reference/beta/resources/horizontalsection.md [hunk 24]
```diff
@@ -21,8 +21,11 @@ Inherits from [entity](../resources/entity.md).
 
 | Method                                                      | Return type                                                       | Description                                                                                               |
 | :---------------------------------------------------------- | :---------------------------------------------------------------- | :-------------------------------------------------------------------------------------------------------- |
-| [List horizontalSections](../api/horizontalsection-list.md) | [horizontalSection](../resources/horizontalsection.md) collection | Get a list of the [horizontalSection](../resources/horizontalsection.md) objects and their properties.    |
+| [Create horizontalSection](../api/sitepage-post-horizontalsection.md)    | [horizontalSection](../resources/horizontalsection.md)            | Create a new [horizontalSection](../resources/horizontalsection.md) object. |
```
Comment (Lauragra) [inline api-reference/beta/resources/horizontalsection.md:24]: We should list the operations in this order: List, Create, Get, Update, Delete.

File: api-reference/beta/resources/verticalsection.md [hunk 25]
```diff
@@ -18,25 +18,32 @@ Represents the vertical section in a given SharePoint page.
...
 
 ## Methods
-|Method|Return type|Description|
-|:---|:---|:---|
-|[Get verticalSection](../api/verticalsection-get.md)|[verticalSection](../resources/verticalsection.md)|Read the properties and relationships of a [verticalSection](../resources/verticalsection.md) object.|
-|[List webParts](../api/webpart-list.md)|[webPart](../resources/webPart.md) Collection |Get a list of web parts associated with a [verticalSection](../resources/verticalSection.md) object.|
 
+| Method                                               | Return type                                        | Description                                                                                           |
+| :--------------------------------------------------- | :------------------------------------------------- | :---------------------------------------------------------------------------------------------------- |
+| [Create verticalSection](../api/sitepage-post-verticalsection.md)    | [verticalSection](../resources/verticalsection.md)            | Create a new [verticalSection](../resources/verticalsection.md) object. |
```
Comment (Lauragra) [inline api-reference/beta/resources/verticalsection.md:24]: Update the order here too: List, Create, Get, Update, Delete.

File: api-reference/beta/toc.yml [hunk 26]
```diff
@@ -16401,6 +16401,12 @@ items:
             href: api/horizontalsection-list.md
           - name: Get
             href: api/horizontalsection-get.md
+          - name: Create
```
Comment (Lauragra) [inline api-reference/beta/toc.yml:16404]: Please update here too for consistency - move the Create before the Get entry in both sections.

**Pull request merged**